src/panels/logbook/ha-panel-logbook.js

```diff
@@ -75,14 +103,38 @@ class HaPanelLogbook extends LocalizeMixin(PolymerElement) {
           alt="[[localize('ui.common.loading')]]"
         ></paper-spinner>
 
-        <vaadin-date-picker
-          id='picker'
-          value='{{_currentDate}}'
-          label="[[localize('ui.panel.logbook.showing_entries')]]"
-          disabled='[[isLoading]]'
-          required
-        ></vaadin-date-picker>
-
+        <div class="flex layout horizontal wrap">
+          <vaadin-date-picker
+            id='picker'
+            value='{{_currentDate}}'
+            label="[[localize('ui.panel.logbook.showing_entries')]]"
+            disabled='[[isLoading]]'
+            required
+          ></vaadin-date-picker>
+  
+          <paper-dropdown-menu
+            label-float
+            label="[[localize('ui.panel.logbook.period')]]"
+            disabled='[[isLoading]]'
+          >
+            <paper-listbox
+              slot="dropdown-content"
+              selected="{{_periodIndex}}"
+            >
+              <paper-item>[[localize('ui.duration.day', 'count', 1)]]</paper-item>
+              <paper-item>[[localize('ui.duration.day', 'count', 3)]]</paper-item>
+              <paper-item>[[localize('ui.duration.week', 'count', 1)]]</paper-item>
+            </paper-listbox>
+          </paper-dropdown-menu>
+  
+          <ha-entity-picker
+            hass="[[hass]]"
+            value="{{_entityId}}"
+            label="[[localize('ui.components.entity.entity-picker.entity')]]"
+            disabled='[[isLoading]]'
+            onChange="{{entityPicked}}"
```

Comment:
`on-change`

Reply:
No, HaEntityPicker works only with onChange

Reply:
You must have looked at the Preact components. https://github.com/home-assistant/home-assistant-polymer/blob/master/src/panels/dev-service/ha-panel-dev-service.js#L110-L117